src/Mvc/Mvc.Core/ref/Microsoft.AspNetCore.Mvc.Core.netcoreapp3.0.cs

```diff
@@ -14,6 +14,7 @@ public static partial class ControllerEndpointRouteBuilderExtensions
         public static Microsoft.AspNetCore.Builder.ControllerActionEndpointConventionBuilder MapControllerRoute(this Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints, string name, string pattern, object defaults = null, object constraints = null, object dataTokens = null) { throw null; }
         public static Microsoft.AspNetCore.Builder.ControllerActionEndpointConventionBuilder MapControllers(this Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints) { throw null; }
         public static Microsoft.AspNetCore.Builder.ControllerActionEndpointConventionBuilder MapDefaultControllerRoute(this Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints) { throw null; }
+        public static void MapDynamicControllerRoute<TTransformer>(this Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints, string pattern) where TTransformer : Microsoft.AspNetCore.Mvc.Routing.DynamicRouteValueTransformer { }
```

Comment:
Should this return `IEndpointRouteBuilder` like the rest

Reply:
**TLDR:** no because it's not useful

Everywhere else we have *dynamic* behaviour in routing, we don't expose a builder because there isn't anything useful you can do with it.

You might say.... hey, I want to use `RequireHost(...)` - that seems cool and good! It won't work, because by the time we're running policies we're looking at the *real* endpoint, not the one used to enable this functionality. So if you put `HostAttribute` on your action and then used a dynamic controller route to select it, then that would work, and it's **super** by-design that it works that way.